crates/relay/src/lib.rs

```diff
@@ -241,13 +245,92 @@ pub mod pallet {
                 reports.insert(vault_id.clone());
             });
 
-            // reward the participant by increasing their SLA
+            // reward the reporter by increasing their SLA
             ext::sla::event_update_vault_sla::<T>(&signer, ext::sla::Action::TheftReport)?;
-
             Self::deposit_event(<Event<T>>::VaultTheft(vault_id, tx_id));
 
             Ok(().into())
         }
+
+        /// Report Vault double payment, providing two fraud proofs (malicious Bitcoin transactions
+        /// and the corresponding transaction inclusion proofs).
+        ///
+        /// # Arguments
+        ///
+        /// * `origin`: Any signed user.
+        /// * `vault_id`: The account of the vault to check.
+        /// * `raw_merkle_proofs`: The proofs of tx inclusion.
+        /// * `raw_txs`: The raw Bitcoin transactions.
+        #[pallet::weight(<T as Config>::WeightInfo::report_vault_theft())]
+        #[transactional]
+        pub fn report_vault_double_payment(
+            origin: OriginFor<T>,
+            vault_id: T::AccountId,
+            raw_merkle_proofs: (Vec<u8>, Vec<u8>),
+            raw_txs: (Vec<u8>, Vec<u8>),
+        ) -> DispatchResultWithPostInfo {
+            ext::security::ensure_parachain_status_not_shutdown::<T>()?;
+            let signer = ensure_signed(origin)?;
+
+            ensure!(
+                raw_merkle_proofs.0 != raw_merkle_proofs.1,
+                Error::<T>::DuplicateMerkleProof,
```

Comment:
I think this check is superfluous if we check for transaction uniqueness, and verify their inclusions using the merkle proofs